content/v2.0/application/settings.md

```diff
@@ -0,0 +1,266 @@
+
+```shell
+ANALYTICS_ENABLED=true MAX_CART_ITEMS=100 bundle exec hanami console
+
+bookshelf[development]> Hanami.app["settings"].analytics_enabled
+=> true
+
+bookshelf[development]> Hanami.app["settings"].max_cart_items
+=> 100
+```
+
+Common types that are useful in settings constructors include:
+
+```ruby
+Types::String
+Types::Params::Bool
+Types::Params::Integer
+Types::Params::Float
+Types::Params::Date
+Types::Params::Time
+```
+
+## Required and optional settings
+
+Whether or not each setting is required for your application to start is determined by its constructor.
+
+If a setting uses a constructor like `Types::String` or `Types::Params::Bool`, then Hanami will raise an exception if the setting is missing, rather than let your application start in a potentially invalid state.
+
+The below settings configuration will result in a `Hanami::Settings::InvalidSettingsError` when `DATABASE_URL` and `ANALYTICS_ENABLED` environment variables are not set:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool
+    setting :max_cart_items, constructor: Types::Params::Integer
+  end
+end
+```
+
+```shell
+bundle exec hanami server
+
+! Unable to load application: Hanami::Settings::InvalidSettingsError: Could not initialize settings. The following settings were invalid:
+
+analytics_enabled:  cannot be coerced to false
+max_cart_items: can’t convert nil into Integer
+```
+
+The same exception will be raised if a setting can't be correctly coerced:
+
+```shell
+ANALYTICS_ENABLED=true MAX_CART_ITEMS="not coerceable to integer"  bundle exec hanami server
+
+! Unable to load application: Hanami::Settings::InvalidSettingsError: Could not initialize settings. The following settings were invalid:
+
+max_cart_items: invalid value for Integer(): "not coerceable to integer"
+```
+
+To make a setting optional, use `optional` to allow `nil` values:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool.optional
+    setting :max_cart_items, constructor: Types::Params::Integer.optional
+  end
+end
+```
+
+## Default values
+
+Settings can also specify defaults to be used in the absence of the relevant environment variable.
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :redis_url, default: "redis://localhost:6379", constructor: Types::String
+    setting :analytics_enabled, default: false, constructor: Types::Params::Bool
+    setting :max_cart_items, default: 100, constructor: Types::Params::Integer
+  end
+end
+```
+
+## Constraints
+
+To enforce additional contraints on settings, you can use a [dry-types](https://dry-rb.org/gems/dry-types) constraint in your constructor.
```

Comment:
If we introduce dry-types earlier, then we can instead link directly to dry-types' constraint docs here:

```suggestion
To enforce additional contraints on settings, you can use a [constraint](https://dry-rb.org/gems/dry-types/1.2/constraints/) in your constructor type.
```